pyat/at/physics/fastring.py

```diff
@@ -100,3 +101,136 @@ def fast_ring(ring: Lattice, split_inds=[]) -> Tuple[Lattice, Lattice]:
                          split_inds=split_inds,
                          detuning_elem=detuning_elem)
     return fastringnorad, fastringrad
+
+
+def simple_ring(energy: float, circumference: float, harmonic_number: int,
+                Qx: float, Qy: float, Vrf: float, alpha: float,
+                beta_x: Optional[float]=1.0, beta_y: Optional[float]=1.0,
+                alpha_x: Optional[float]=0.0, alpha_y: Optional[float]=0.0,
+                Qpx: Optional[float]=0.0, Qpy: Optional[float]=0.0,
+                A1: Optional[float]=0.0, A2: Optional[float]=0.0,
+                A3: Optional[float]=0.0, emit_x: Optional[float]=0.0,
+                emit_y: Optional[float]=0.0, sigma_dp: Optional[float]=0.0,
+                tau_x: Optional[float]=0.0, tau_y: Optional[float]=0.0,
+                tau_z: Optional[float]=0.0, U0: Optional[float]=0.0,
+                SetTimeLag: Optional[bool]=False
+                ):
+    """Generates a "simple ring" based on a given dictionary
+       of global parameters
+
+    A simple ring consists of:
+
+    * an RF cavity,
+    * a 6x6 linear transfer map,
+    * a detuning and chromaticity element,
+    * a simplified quantum diffusion element
+        which contains equilibrium emittance and radiation damping
+
+    Positional Arguments:
+        * energy [eV]
+        * circumference [m]
+        * harmonic_number - can be scalar or 1d array
+        * Qx - horizontal tune
+        * Qy - vertical tune
+        * Vrf - RF Voltage set point [V] - can be scalar or 1d array
+        * alpha (momentum compaction factor)
+
+    Optional Arguments:
+        * beta_x, beta_y [m]
+        * alpha_x, alpha_y
+        * Qpx, Qpy - linear chromaticities
+        * A1, A2, A3 - amplitude detuning coefficients
+        * emit_x, emit_y, sigma_dp - equilibrium values [m.rad, m.rad, -]
+        * tau_x, tau_y, tau_z - radiation damping times [turns]
+        * U0 - energy loss [eV] (positive number)
+        * SetTimeLag - Boolean to determine if TimeLag of cavity should be set
+
+    If the given emit_x,emit_y or sigma_dp is 0, then no equlibrium emittance
+    is applied in this plane.
+    If the given tau is 0, then no radiation damping is applied for this plane.
+
+
+    Returns:
+        ring (Lattice):    Simple ring
+    """
+
+    # compute slip factor
+    gamma = energy / e_mass
+    eta = alpha - 1/gamma**2
+
+    harmonic_number = numpy.atleast_1d(harmonic_number)
+    Vrf = numpy.atleast_1d(Vrf)
+
+    assert len(harmonic_number) == len(Vrf), (
+        "harmonic_number input must match length of Vrf input"
+        )
+
+    # compute rf frequency
+    frf = harmonic_number * clight / circumference
+
+    all_cavities = []
+    for icav in numpy.arange(len(harmonic_number)):
```

Comment:
There is a mistake there, Timelag applies to all cavities and should be computed outside of the loop using Vrf[0] and frf[0].
However this is not safe: it is said nowhere that your main RF system should be at index 0. You have to improve the help and check that harmonic numbers are properly sorted. Other question, how to you set the Timelag of your harmonic cavity to tune its phase w.r.t. the main RF?
A Timelag input similar to Vrf and harmonic number (scalar or array) seems necessary... in this case do you really want to add this calculation inside the function? It looks like an unnecessary complication especially since you can use the set_cavity_phase function after your ring is created, I would just remove this part.

Reply:
done